packages/next/next-server/lib/router/router.ts

```diff
@@ -318,14 +343,12 @@ export default class Router implements BaseRouter {
       return
     }
 
+    const { url, as, options } = e.state
```

Comment:
Do we have to worry about null checking `e.state`? It seems like the previous control flow path might of handled an edge case where this was null.

Reply:
It's checked and short-circuited in the conditional right before this line